### PR TITLE
Improve configurable retry logic tests

### PR DESCRIPTION
**Pull request by DavoudEshtehari**

### CODE REVIEW

**Review by cheenamalhotra — APPROVED** [review]
LGTM! Just a few minor touch-ups would be needed.

File: BUILDGUIDE.md [hunk 1]
```diff
@@ -148,6 +148,48 @@ Unix (`netcoreapp`):
...
+```bash
+> dotnet test "src\Microsoft.Data.SqlClient\tests\PerformanceTests\Microsoft.Data.SqlClient.Performance.Tests.csproj" /p:Platform="AnyCPU" /p:Configuration="Release" /p:TestTargetOS="Windowsnetcoreapp" --no-build -v n --filter "category!=nonnetcoreapptests&category!=failing&category!=nonwindowstests"
+```
+
+Unix (`netcoreapp`):
+```bash
+> dotnet test "src/Microsoft.Data.SqlClient/tests/PerformanceTests/Microsoft.Data.SqlClient.Performance.Tests.csproj" /p:Platform="AnyCPU" /p:Configuration="Release" /p:TestTargetOS="Unixnetcoreapp" --no-build -v n --filter "category!=nonnetcoreapptests&category!=failing&category!=nonlinuxtests&category!=nonuaptests"
+```
+
+## Run A Single Test
```
Comment (cheenamalhotra) [inline BUILDGUIDE.md:188]: You could also include a section on how to collect benchmarks.

File: src/Microsoft.Data.SqlClient.sln [hunk 2]
```diff
@@ -172,6 +172,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Data.SqlClient.Te
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Data.SqlClient.ExtUtilities", "Microsoft.Data.SqlClient\tests\tools\Microsoft.Data.SqlClient.ExtUtilities\Microsoft.Data.SqlClient.ExtUtilities.csproj", "{E4C08DCE-DC29-4FEB-B655-1E7287DB5A2B}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Data.SqlClient.Performance.Tests", "Microsoft.Data.SqlClient\tests\PerformanceTests\Microsoft.Data.SqlClient.Performance.Tests.csproj", "{303FB396-6917-4153-B6A6-16CC5E24768B}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ConfigurableRetryLogic.Sample", "Microsoft.Data.SqlClient\tests\CustomConfigurableRetryLogic\ConfigurableRetryLogic.Sample.csproj", "{B499E477-C9B1-4087-A5CF-5C762D90E433}"
```
[the remark this answers is not in the document]
Comment (DavoudEshtehari) [reply src/Microsoft.Data.SqlClient.sln:177]: Changing to "CustomRetryLogicProvider".

File: BUILDGUIDE.md (hunk 1 above)
Comment (DavoudEshtehari) [reply BUILDGUIDE.md:188]: Good idea. But don't you think it'd be better to add a section in ADO.NET documents for this purpose? Then we can put its link here.

Comment (cheenamalhotra) [reply BUILDGUIDE.md:188]: It's our test suite.. ADO.NET docs are for driver documentations.. so no.. we document things only here for test lab building.

**Review by cheenamalhotra — CHANGES_REQUESTED** [review]
EDIT: The project goal is to calculate benchmarks which seems to be missing as of now. Deferred to next milestone until further changes are made.

**Pull request closed without merging**